Makefile

```diff
@@ -280,6 +281,7 @@ generate-go-deepcopy: $(CONTROLLER_GEN) ## Generate deepcopy go code for core
 .PHONY: generate-modules
 generate-modules: ## Run go mod tidy to ensure modules are up to date
```

Comment:
It's highly likely that we have to introduce a dependabot action (see pr-dependabot.yaml from core CAPI).

But I'm fine waiting for the first time a dependabot PR fails with modules being out of sync and then only adding it if actually needed (it's very likely as we needed it in core CAPI & CR as well as soon as we had multiple modules referencing each other)